### PR TITLE
Task4

### PR DESCRIPTION
**Pull request by Aksana-Tsishchanka**

### CODE REVIEW

File: data/index.html [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<!DOCTYPE html>
```
Comment (boccob) [inline data/index.html]: It's not data :)

File: http-servers/html-server.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  
+  stream.on('error', function() {
+    res.writeHead(404, 'Not Found');
+    res.end('404: File Not Found!');
+  });
+  
+  res.writeHead(200, {
+    'Content-Type': 'text/html'
+  });
+  stream.pipe(res);
```
Comment (boccob) [inline http-servers/html-server.js]: It seems that you don't replace message

File: middlewares/index.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  if (req.headers.cookie) {
+    req.parsedCookies = getCookie(req.headers.cookie);
+  } else req.parsedCookies = {};
+  req.pipe(res);
+  next();
+};
+
+export const queryParser = function(req, res, next) {
+  req.parsedQuery = req.query;
+  req.pipe(res);
```
Comment (boccob) [inline middlewares/index.js]: But why?

File: middlewares/index.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+export const cookieParser = function(req, res, next) {
+  if (req.headers.cookie) {
+    req.parsedCookies = getCookie(req.headers.cookie);
+  } else req.parsedCookies = {};
+  req.pipe(res);
+  next();
+};
+
+export const queryParser = function(req, res, next) {
+  req.parsedQuery = req.query;
```
Comment (boccob) [inline middlewares/index.js:19]: Great :)

File: app.js [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+  }
+  products.push(product);
+}
+
+function getModelById(models, id) {
+  return models.filter(model => model.id == id)[0];
+}
+
+app
+  .use(bodyParser.json())
```
Comment (boccob) [inline app.js]: Better to support x-www-form-urlencoded as well. But it's just another line with bodyParser.

File: data/index.html [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+<html>
+    <head></head>
+    <style type="text/css">
```
Comment (boccob) [inline data/index.html]: Interesting place for style tag.

File: app.js [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+  .use(cookieParser)
+  .use(queryParser)
+  .use('/', router);
+
+router.get('/', (request, response) => {
```
Comment (boccob) [inline app.js:34]: Does it work for you?

File: middlewares/index.js [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+  let result = {};
+  let arr = cookie.split(';');
+  arr.map(el => {
+    const cookieStr = el.split('=');
+    result[cookieStr[0]] = cookieStr[1];
+  });
+  return result;
+}
+
+export const cookieParser = function(req, res, next) {
```
Comment (boccob) [inline middlewares/index.js:11]: Please check keys of result array. They have redundant space inside.

File: http-servers/json-server.js [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+  ]
+};
+
+const server = http.createServer((req,res) => {
+  req.on('error', (err) => {
+    console.error(err.stack);
+  });
+  
+  res.setHeader('Content-Type', 'application/json');
+  res.write(JSON.stringify(product));
```
Comment (boccob) [inline http-servers/json-server.js]: It's not allowed to use res.write for this part of task :)

File: app.js (hunk 5 above)
Comment (Aksana-Tsishchanka) [reply app.js]: added, thx

File: data/index.html (hunk 1 above)
Comment (Aksana-Tsishchanka) [reply data/index.html]: Moved to https-servers/serve/index.html

File: http-servers/html-server.js (hunk 2 above)
Comment (Aksana-Tsishchanka) [reply http-servers/html-server.js]: fixed, but not for streams(

File: http-servers/json-server.js (hunk 9 above)
Comment (Aksana-Tsishchanka) [reply http-servers/json-server.js]: I didn't find another way how send json

File: data/index.html (hunk 6 above)
Comment (Aksana-Tsishchanka) [reply data/index.html]: fixed, thx

File: middlewares/index.js (hunk 8 above)
Comment (Aksana-Tsishchanka) [reply middlewares/index.js:11]: added

File: middlewares/index.js (hunk 3 above)
Comment (Aksana-Tsishchanka) [reply middlewares/index.js]: sorry, not clear what is wrong

File: app.js (hunk 7 above)
Comment (Aksana-Tsishchanka) [reply app.js:34]: yes http://take.ms/Ksw1o

**Pull request merged**